### PR TITLE
VIH-6620 fix for acceptance  test no answers

### PR DESCRIPTION
**Pull request by valentinabennett**
### JIRA link (if applicable) ###

VIH-6620

### Change description ###

Fix for acceptance test, check on null answers ( in case no answers)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by npoxon — APPROVED** [review]

**Review by jdssunil — APPROVED** [review]

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Hooks/DataHooks.cs [hunk 1]
```diff
@@ -100,8 +100,8 @@ private void UserShouldNotHaveAnswers(TestApiManager api)
         {
             var response = api.GetSuitabilityAnswers(_username);
             var answers = RequestHelper.Deserialise<List<PersonSuitabilityAnswerResponse>>(response.Content);
-            answers.Count.Should().Be(1, $"user with username '{_username}' has {answers.Count} previous answer(s) from other hearings saved");
-            if (answers.First().Answers.Count > 0)
+
+            if (answers != null && answers.First() != null && answers.First().Answers.Count > 0)
```
Comment (levibutler) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Hooks/DataHooks.cs:104]: could use `?` to check for null
`answers?.First()`

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Hooks/RemoveDataHooks.cs [hunk 2]
```diff
@@ -52,7 +52,7 @@ private void UserShouldNotHaveAnswers(TestApiManager api)
             var response = api.GetSuitabilityAnswers(_username);
             var answers = RequestHelper.Deserialise<List<PersonSuitabilityAnswerResponse>>(response.Content);
 
-            if (answers.Count > 0)
+            if ( answers != null && answers.Count > 0)
```
Comment (levibutler) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Hooks/RemoveDataHooks.cs:55]: could be
`if (answers?.Count > 0)`
:)

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/AnswersSteps.cs [hunk 3]
```diff
@@ -101,7 +101,7 @@ private bool AnswersHaveBeenSaved()
             var response = _c.Api.GetSuitabilityAnswers(_c.CurrentUser.Username);
             var answers = RequestHelper.Deserialise<List<PersonSuitabilityAnswerResponse>>(response.Content);
 
-            if (!answers.Any(x => x.Hearing_id.Equals(_c.Test.Hearing.Id)))
```
Comment (levibutler) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/AnswersSteps.cs:104]: we could simplify to

```
var hearingAnswer = answers?.FirstOrDefault(a => a.Hearing_id.Equals(_c.Test.Hearing.Id);
return hearingAnswer?.Answers.Any() ?? false
```

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Hooks/DataHooks.cs (hunk 1 above)
Comment (valentinabennett) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Hooks/DataHooks.cs:104]: done

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/AnswersSteps.cs (hunk 3 above)
Comment (valentinabennett) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/AnswersSteps.cs:104]: done

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Hooks/RemoveDataHooks.cs (hunk 2 above)
Comment (valentinabennett) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Hooks/RemoveDataHooks.cs]: done

File: ServiceWebsite/ServiceWebsite.AcceptanceTests/Hooks/DataHooks.cs [hunk 4]
```diff
@@ -100,8 +100,8 @@ private void UserShouldNotHaveAnswers(TestApiManager api)
         {
             var response = api.GetSuitabilityAnswers(_username);
             var answers = RequestHelper.Deserialise<List<PersonSuitabilityAnswerResponse>>(response.Content);
-            answers.Count.Should().Be(1, $"user with username '{_username}' has {answers.Count} previous answer(s) from other hearings saved");
-            if (answers.First().Answers.Count > 0)
+
+            if (answers?.First() != null && answers.First().Answers.Count > 0)
```
Comment (levibutler) [inline ServiceWebsite/ServiceWebsite.AcceptanceTests/Hooks/DataHooks.cs:104]: should this be first or default if doing null check?
could actually be done with just 
`answers?.FirstOrDefault()?.Answers.Count > 0`

Comment (valentinabennett) [reply ServiceWebsite/ServiceWebsite.AcceptanceTests/Hooks/DataHooks.cs]: Done

**Review by levibutler — APPROVED** [review]

**Pull request merged**